File: 0055-jump-game/0055-jump-game.cpp
#include<vector>
using namespace std;

class Solution {
public:
    bool canJump(vector<int>& nums) {
        int lastGoodIndex = nums.size() - 1;
        for (int i = nums.size() - 1; i >= 0; i--) {
            if (i + nums[i] >= lastGoodIndex) {
                lastGoodIndex = i;
            }
        }
        return lastGoodIndex == 0;
    }
};

File: 0055-jump-game/0055-jump-game.py
from typing import List

class Solution:
    def canJump(self, nums: List[int]) -> bool:
        last_good_index = len(nums) - 1
        for i in reversed(range(len(nums))):
            if i + nums[i] >= last_good_index:
                last_good_index = i
        return last_good_index == 0
